### ISSUE TITLE
[Bug] Unable to start in Docker Mode

### ISSUE THREAD

**Issue by haxpak** [cobbr/Covenant#161]
I just ran

```
$ ~ > git clone --recurse-submodules https://github.com/cobbr/Covenant
$ ~ > cd Covenant/Covenant
$ ~/Covenant/Covenant > docker build -t covenant .
```

the docker build is complete 
however when i try to start the docker with 

```console
> docker run -it -p 7443:7443 -p 127.0.0.1:2001:80 -p 127.0.0.1:2000:443 --name covenant -v /home/user/covenant-data:/app/Data covenant
```

I get the following error
```
Error: git submodules have not been initialized
Covenant's submodules can be cloned with: git clone --recurse-submodules https://github.com/cobbr/Covenant
Or initialized after cloning with: git submodule update --init --recursive

```

can someone give me a fix?

**Comment by Hach0u** [comment]
I have the same issue. I'm trying to figure it out from where it's coming from.

**Comment by Hach0u** [comment]
@haxpak, I managed to make it work by adding a proxy to the DockerFile. It's a bit surprising as we already used the --recurse-submodules.

**Comment by cobbr** [comment]
Sorry, just seeing this issue. It looks to me like you are not using the correct path to the Data folder.

Something like this seems more appropriate:

```
> docker run -it -p 7443:7443 -p 127.0.0.1:2001:80 -p 127.0.0.1:2000:443 --name covenant -v /home/user/Covenant/Covenant/Data:/app/Data covenant
```

**Comment by Hach0u** [comment]
Thanks @cobbr . I double checked the volume path, it's correct. Anyway, I'm able to make it works now.

**Comment by sancao2** [comment]
I didnt make it. @Hach0u, how do u add the proxy into Dockerfile?

**Comment by Hach0u** [comment]
Hi @sancao2, I did a quick search on Google => [https://docs.docker.com/network/proxy/](url)

**Comment by H8to** [comment]
> Thanks @cobbr . I double checked the volume path, it's correct. Anyway, I'm able to make it works now.

Hey Hanch0u,
For reasons of completeness could you point out your changes to make it work and close the issue? Thanks!

**Comment by cobbr** [comment]
Hey @Hach0u , I'm not really sure I understand this issue fully. Would you mind describing for us what exactly this issue was and how you fixed it?

**Comment by Hach0u** [comment]
Hi @cobbr & @H8to,
I had the same issue as @haxpak, which is 
`Error: git submodules have not been initialized`
From my understanding, it's because the container is not able to access properly submodule files that are mounted through the volume. I have a "tricky" docker configuration that I do not fully understand :D, so it's just supposition. 
To make it work, I added proxy config in the Dockerfile. I guess that submodules are automatically downloaded while the container is created/started and not during the git clone. It solved my issue (probably not the root cause but as a workaround).
To sum up, I think it's an access right issue between the container and the mounted volume, nothing related to Covenant.

**Comment by cobbr** [comment]
Thanks for the info @Hach0u . Closing since the issue was with a custom Dockerfile and no response from @haxpak .

**Comment by royharoush** [comment]
happensd to me now, with the exact same scenario\flow of docker
